src/compas_model/algorithms/collider.py

```diff
@@ -0,0 +1,357 @@
+        if not all(fabs(point[2]) < tmax for point in projected):
+            # for point in projected:
+            #     print("tolerance is off " + str(fabs(point[2])) + " > " + str(tmax))
+            print("to_shapely_polygon: tolerance is off")
+            return None
+        elif polygon.area < amin:
+            print("area is off " + str(polygon.area) + " < " + str(amin))
+            # print("area is off " + str(polygon.area) + " < " + str(amin))
+            print("to_shapely_polygon: area is off")
+            return None
+        else:
+            return ShapelyPolygon(projected)
+
+    def to_compas_polygon(matrix, shapely_polygon):
+        """convert a shapely polygon to compas polygon back to the frame"""
+
+        # convert coordiantes to 3D by adding the z coordinate
+        coords = [[x, y, 0.0] for x, y, _ in shapely_polygon.exterior.coords]
+
+        # orient points to the original first mesh frame
+        coords = transform_points(coords, matrix.inverted())[:-1]
+
+        # convert to compas polygon
+        return Polygon(coords)
+
+    def is_coplanar(frame0, frame1, t_normal_colinearity=1e-1, t_dist_frames=1e-1):
+        """check if two frames are coplanar and at the same position"""
+
+        # get the normal vector of the first frame
+        normal0 = frame0.normal
+
+        # get the normal vector of the second frame
+        normal1 = frame1.normal
+
+        cross_product = normal0.cross(normal1)
+
+        # check if the two normal vectors are parallel
+        are_parellel = abs(cross_product.length) < t_normal_colinearity
+
+        # are planes at the same positions?
+        plane = Plane(frame0.point, frame0.normal)
+        projected_point = plane.projected_point(frame1.point)
+        are_close = distance_point_point(projected_point, frame1.point) < t_dist_frames
+        return are_parellel and are_close
+
+    def get_average_frame(polygon):
+        # --------------------------------------------------------------------------
+        # number of points
+        # --------------------------------------------------------------------------
+        points = list(polygon.points)
+        n = len(points)
+
+        # --------------------------------------------------------------------------
+        # origin
+        # --------------------------------------------------------------------------
+        origin = Point(0, 0, 0)
+        for point in points:
+            origin = origin + point
+        origin = origin / n
+
+        # --------------------------------------------------------------------------
+        # xaxis
+        # --------------------------------------------------------------------------
+        xaxis = points[1] - points[0]
+        xaxis.unitize()
+
+        # --------------------------------------------------------------------------
+        # zaxis
+        # --------------------------------------------------------------------------
```

Comment:
and please remove all these elaborate comments. the code is already clear without them...

Reply:
removed